redis/src/main/java/org/axonframework/redis/eventhandling/tokenstore/repository/DefaultRedisTokenRepository.java

```diff
@@ -0,0 +1,161 @@
+import static org.axonframework.redis.eventhandling.tokenstore.repository.DefaultRedisTokenScripts.RELEASE_TOKEN_SCRIPT;
+import static org.axonframework.redis.eventhandling.tokenstore.repository.DefaultRedisTokenScripts.RELEASE_TOKEN_SHA1;
+import static org.axonframework.redis.eventhandling.tokenstore.repository.DefaultRedisTokenScripts.STORE_TOKEN_SCRIPT;
+import static org.axonframework.redis.eventhandling.tokenstore.repository.DefaultRedisTokenScripts.STORE_TOKEN_SHA1;
+
+/**
+ * Default implementation of the RedisTokenRepository. Depends on a JedisPool.
+ *
+ * @author Michael Willemse
+ */
+public class DefaultRedisTokenRepository implements RedisTokenRepository {
+
+    private final JedisPool jedisPool;
+
+    private static final Long TRUE = 1L;
+
+    /**
+     * Initialize the RedisTokenRepository with a {@code jedisPool}.
+     * @param jedisPool Configured JedisPool instance
+     */
+    public DefaultRedisTokenRepository(JedisPool jedisPool) {
+        this.jedisPool = jedisPool;
+    }
+
+    @Override
+    public boolean storeTokenEntry(RedisTokenEntry tokenEntry, Instant expirationFromTimestamp) {
+        Long value = (Long) tryEvalSha1(STORE_TOKEN_SCRIPT, STORE_TOKEN_SHA1,
+                getHashKeyList(tokenEntry.getProcessorName(), tokenEntry.getSegment()),
+                Arrays.asList(
+                        tokenEntry.getProcessorName(),
+                        Integer.toString(tokenEntry.getSegment()),
+                        tokenEntry.getOwner(),
+                        Long.toString(tokenEntry.timestamp().toEpochMilli()),
+                        Long.toString(expirationFromTimestamp.toEpochMilli()),
+                        Base64.getEncoder().encodeToString(tokenEntry.getSerializedToken().getData()),
+                        tokenEntry.getSerializedToken().getType().getName()));
+
+        return Objects.equals(value, TRUE);
+    }
+
+    @Override
+    public RedisTokenEntry fetchTokenEntry(String processorName, int segment, String owner, Instant currentTimestamp, Instant expirationFromTimestamp) {
+
+        String redisOutput = (String) tryEvalSha1(FETCH_TOKEN_SCRIPT, FETCH_TOKEN_SHA1,
+                getHashKeyList(processorName, segment),
+                Arrays.asList(
+                        processorName,
+                        Integer.toString(segment),
+                        owner,
+                        Long.toString(currentTimestamp.toEpochMilli()),
+                        Long.toString(expirationFromTimestamp.toEpochMilli())));
+
+        if (redisOutput != null) {
+            JSONObject tokenEntry = new JSONObject(redisOutput);
+
+            String tokenEntryProcessorName = null;
+            int tokenEntrySegment = 0;
+            String tokenEntryOwner = null;
+            String tokenEntryTimestamp = null;
+            byte[] tokenEntryToken = null;
+            String tokenEntryTokenType = null;
+
+            if (tokenEntry.has("processorName")) {
```

Comment:
Small adjustment, but might be nice to change these Strings (e.g. `processorName`/`segment`/...) into private constants.